packages/appium/docs/en/ecosystem/build-docs.md

```diff
@@ -42,32 +42,178 @@ The other main piece is automatic documentation generation from your code files.
+
+##### JavaScript
+
+```js
+// note: this is equivalent to the TypeScript example below
+
+class MyExtension {
+  static newMethodMap = /** @type {const} */({
+    // ...
+  });
+  
+  static executeMethodMap = /** @type {const} */({
+    // ...
+  });
+}
+
+```
+
+##### TypeScript
+
+```ts
+class MyExtension {
+  static newMethodMap = {
+    // ...
+  } as const;
+
+  static executeMethodMap = {
+    // ...
+  } as const;
+}
+```
+
+#### Driver Constraints
+
+> Note: Plugin authors can skip this section.
+
+Your driver may have a property `desiredCapConstraints`. It should also follow the same pattern as `newMethodMap` and `executeMethodMap` above.  For example:
+
+```js
+class MyExtension {
+  desiredCapConstraints = /** @type {const} */({
+    myCapability: {
+      presence: true,
+      isString: true
+    },
+    myOtherCap: {
+      isBoolean: true
+    }
+  });
+}
+```
+
+(For extensions written in TypeScript, use `as const` as before.)
+
+#### Commands
+
+The documentation for a command, as defined in your extension, comes from multiple places. These sources are then combined as needed into the final output.
+
+In Appium, new commands are defined in `newMethodMap` and execute methods are defined in `executeMethodMap`.  **The value of these properties are used to build your documentation.**  In particular, parameter names and optional/required status _override_ whatever method implementation does.  So for example, if your `newMethodMap` contains:
+
+```js
+class MyExtension {
+  static newMethodMap = /** @type {const} */({
+    '/session/:sessionId/myThing': {
+      /**
+       * Does my thing
+       */
+      GET: {command: 'doMyThing', payloadParams: {required: ['a', 'b']}},    
+    }
+  })
+}
+```
+
+...and your `doMyThing` method implementation looks like this:
+
+```js
+class MyExtension {
+  /**
+   * Doesn't do my thing
+   * @param {any} a - Whatever
+   * @param {number} d - Some number
+   * @param {boolean} c - Some boolean
+   * @returns {Promise<boolean>} Some other boolean
+   */
+  async doMyThing(a, d = 1, c = false) {
+    // ...
+    return true;
+  }
+}
+```
+
+The documentation will show that the `doMyThing` method accepts required parameters `a` and `b`. Even though `b` is named `d` in the implementation--and it's optional--it will be ignored. Likewise, since the definition in `newMethodMap` knows nothing about `c`, it too is ignored.
+
+In addition, the _description_ from the docstring in `newMethodMap` overrides the description in the method implementation; it will describe the command `doMyThing` as "Does my thing".
+
+Finally, the `@param` tags and `@returns` tag from the docstring in `newMethodMap` provides further information where `newMethodMap` cannot; e.g., each parameter has a type and description.
```

Comment:
something seems off about this sentence. can you double check it says the right thing?